src/librustc_typeck/collect.rs

```diff
@@ -546,16 +552,11 @@ fn convert_field<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
 {
     let tt = ccx.icx(struct_predicates).to_ty(&ExplicitRscope, &field.ty);
     ty_f.fulfill_ty(tt);
-    write_ty_to_tcx(ccx, field.id, tt);
```

Comment:
Correct, it was all a bit of a mess before, but that's all incompatible with per-body tables. Should add this change to the description.